app_dart/lib/src/service/scheduler.dart

```diff
@@ -143,12 +146,67 @@ class Scheduler {
       return;
     }
 
+    final slug = pr.base!.repo!.slug();
     final String fullRepo = pr.base!.repo!.fullName;
     final String? branch = pr.base!.ref;
     final String sha = pr.mergeCommitSha!;
 
     final String id = '$fullRepo/$branch/$sha';
     final Key<String> key = datastore.db.emptyKey.append<String>(Commit, id: id);
+
+    // In non-merge queue cases, the `mergedAt` time for closed pull request was
+    // enough to correctly identify the order of linear history.
+    //
+    // Dates are hard to reason in the merge queue. Commits can get held up in queue
+    // by slower running tests ahead of them. When this occurs, and all tests pass,
+    // these commits will all be "merged at" the same time. Conversely, Auto Submit
+    // bot can queue up multiple commits at once - each generating a `gh-readonly-queue`
+    // branch. When this occurs, they can have the same "committedDate".
+    //
+    // In either case, this will confuse the dashboard and backfill scheduler. Tree status
+    // can be "red" for no apparent reason (red reports followed by green reports).
+    //
+    // A solution in both cases: use graphql to query the parent sha of the merged commit.
+    // This can be used to disambiguate the commit history when queried later.
+    //
+    // See: https://github.com/flutter/flutter/issues/162830
+    final graphql = await config.createGitHubGraphQLClient();
+    const source = r'''
+query GetFlutterCommitDateAndParent {
+  repository(owner: $sRepoOwner, name: $sRepoName) {
+    object(oid: $sCommitOid) {
+      ... on Commit {
+        parents(first: 1) {  # Get the first parent (usually the only one)
+          nodes {
+              oid
+          }
+        }
+      }
+    }
+  }
+}''';
+    final result = await graphql.query(
```

Comment:
Not worth blocking this PR, but this does introduce an additional place where we can hit a burst quota limit and grind everything to a halt. We need a strategy to deal with this, I'll file a bug.

Reply:
I believe GraphQL falls in another [API bucket](https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api#calculating-points-for-the-secondary-rate-limit): 

> No more than 900 points per minute are allowed for REST API endpoints, and no more than 2,000 points per minute are allowed for the GraphQL API endpoint